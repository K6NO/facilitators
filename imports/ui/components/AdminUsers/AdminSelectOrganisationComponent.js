import React from 'react';
import ReactSelect from 'react-select';
import PropTypes from 'prop-types';
import { Roles } from 'meteor/alanning:roles';

// import './AdminSelectOrganisationComponent.scss';

class AdminSelectOrganisationComponent extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            selectedOrgId: this.props.selectOrganisation,
        }
    };

    componentWillReceiveProps(nextProps) {
        if(nextProps.selectOrganisation !== this.props.selectOrganisation) {
            this.setState({
                selectedOrgId : nextProps.selectOrganisation
            });
        }
    }
    
    handleChange = (selection) => {
        let { selectOrganisationCallback } = this.props;
        this.setState({
            selectedOrgId: selection ? selection._id : '',
        });
        selectOrganisationCallback(selection);
    };

    render() {
        const { organisations, editing, organisationId, user } = this.props;
        const org = organisationId && organisationId.length > 5 
            && organisations.find(org => org._id === organisationId);
        const orgName = org ? org.name : '';

        return (
            editing ? (
                !!Roles.userIsInRole(user._id, ['superadmin']) ? 'Cannot change' :
                <ReactSelect 
                    className="AdminSelectOrganisationComponent"
                    labelKey="name"
                    valueKey="_id"
                    options={organisations}
                    value={this.state.selectedOrgId}
                    resetValue=""
                    name="organisationsSelect"
                    onChange={this.handleChange}/>
            ) : 
                <span>
                    {orgName}
                </span> 
            
        )
    }
}

AdminSelectOrganisationComponent.defaultProps = {
    organisationId: '',
};

AdminSelectOrganisationComponent.propTypes = {
    user: PropTypes.object.isRequired,
    organisations: PropTypes.arrayOf(PropTypes.object).isRequired,
    editing: PropTypes.bool.isRequired,
    organisationId: PropTypes.string
};
  
export default AdminSelectOrganisationComponent;