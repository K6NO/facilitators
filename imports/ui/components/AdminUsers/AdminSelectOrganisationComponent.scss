.AdminSelectOrganisationComponent {
    height: 2.5rem;
    border-radius: 0;
}