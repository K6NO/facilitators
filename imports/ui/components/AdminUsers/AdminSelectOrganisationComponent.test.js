import React from 'react';
import renderer from 'react-test-renderer';
import AdminSelectOrganisationComponent from './AdminSelectOrganisationComponent';

const user = {
    _id: "UserId1"
};
const organisations = [{
    _id: '123',
    name: 'org1name',
    }, {
        _id: '456',
        name: 'org2name',
    }
];
const organisationId = '123'
const editing = true;

test('AdminSelectOrganisationComponent renders in editing and non-editing states', ()=>{
    const mockFunction = jest.fn();
    const component = renderer.create(
        <AdminSelectOrganisationComponent
            editing={editing}
            organisations={organisations}
            selectOrganisationCallback={mockFunction}
            selectedOrganisation={organisationId}
            organisationId={organisationId}
            user={user}
         />

    );
    const nonEditingComponent = 
    renderer.create(
        <AdminSelectOrganisationComponent
            editing={editing}
            organisations={organisations}
            selectOrganisationCallback={mockFunction}
            selectedOrganisation={organisationId}
            organisationId={organisationId}
            user={user}
         />
    );

    const editingTree = component.toJSON();
    const nonEditingTree = nonEditingComponent.toJSON();
    expect(editingTree).toMatchSnapshot();
    expect(nonEditingTree).toMatchSnapshot();
});