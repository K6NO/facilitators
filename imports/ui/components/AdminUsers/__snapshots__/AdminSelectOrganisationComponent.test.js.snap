// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminSelectOrganisationComponent renders in editing and non-editing states 1`] = `
<div
  className="Select AdminSelectOrganisationComponent is-clearable is-searchable Select--single"
>
  <div
    className="Select-control"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="Select-multi-value-wrapper"
      id="react-select-2--value"
    >
      <div
        className="Select-placeholder"
      >
        Select...
      </div>
      <div
        className="Select-input"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <input
          aria-activedescendant="react-select-2--value"
          aria-expanded="false"
          aria-haspopup="false"
          aria-owns=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          role="combobox"
          style={
            Object {
              "boxSizing": "content-box",
              "width": "5px",
            }
          }
          value=""
        />
        <div
          style={
            Object {
              "height": 0,
              "left": 0,
              "overflow": "scroll",
              "position": "absolute",
              "top": 0,
              "visibility": "hidden",
              "whiteSpace": "pre",
            }
          }
        >
          
        </div>
      </div>
    </div>
    <span
      className="Select-arrow-zone"
      onMouseDown={[Function]}
    >
      <span
        className="Select-arrow"
        onMouseDown={[Function]}
      />
    </span>
  </div>
</div>
`;

exports[`AdminSelectOrganisationComponent renders in editing and non-editing states 2`] = `
<div
  className="Select AdminSelectOrganisationComponent is-clearable is-searchable Select--single"
>
  <div
    className="Select-control"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="Select-multi-value-wrapper"
      id="react-select-3--value"
    >
      <div
        className="Select-placeholder"
      >
        Select...
      </div>
      <div
        className="Select-input"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <input
          aria-activedescendant="react-select-3--value"
          aria-expanded="false"
          aria-haspopup="false"
          aria-owns=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          role="combobox"
          style={
            Object {
              "boxSizing": "content-box",
              "width": "5px",
            }
          }
          value=""
        />
        <div
          style={
            Object {
              "height": 0,
              "left": 0,
              "overflow": "scroll",
              "position": "absolute",
              "top": 0,
              "visibility": "hidden",
              "whiteSpace": "pre",
            }
          }
        >
          
        </div>
      </div>
    </div>
    <span
      className="Select-arrow-zone"
      onMouseDown={[Function]}
    >
      <span
        className="Select-arrow"
        onMouseDown={[Function]}
      />
    </span>
  </div>
</div>
`;
